library/core-components/components/button/button.tsx

```diff
@@ -92,9 +97,26 @@ export const RadiusButton = forwardRef<RadiusButtonTag, RadiusButtonProps>(
           ref={element.props.ref}
           {...element.props}
         >
-          <Typography font="--typography-component-typography-button-label">
-            {children}
-          </Typography>
+          <RadiusAutoBox
+            space="--spacing-component-spacing-button-gap"
+            alignment="center"
+            padding="--spacing-component-spacing-button-padding-vertical
+              --spacing-component-spacing-button-padding-horizontal"
+          >
+            <Typography
+              // @ts-expect-error - type error will occur until Typography polymorphic implementation is fixed
+              as="span"
+              font="--typography-component-typography-button-label"
+            >
+              {children}
+            </Typography>
+            {rightIcon && (
+              <RadiusIcon
+                component={rightIcon}
+                size="--sizing-component-sizing-button-icon"
+              />
+            )}
+          </RadiusAutoBox>
```

Comment:
Can we scale back our polymorphic approach for now? It's getting a bit out of hand as these components grow, and is causing issues in a few places 🤔